lib/features/mailbox/data/repository/mailbox_repository_impl.dart

```diff
@@ -48,51 +48,39 @@ class MailboxRepositoryImpl extends MailboxRepository {
       return MailboxResponse(mailboxes: response.first, state: response.last);
     });
 
-    yield localMailboxResponse;
-
-    if (localMailboxResponse.hasData()) {
-      bool hasMoreChanges = true;
-      State? sinceState = localMailboxResponse.state!;
-
-      while(hasMoreChanges && sinceState != null) {
-        final changesResponse = await mapDataSource[DataSourceType.network]!.getChanges(session, accountId, sinceState, properties: properties);
-
-        hasMoreChanges = changesResponse.hasMoreChanges;
-        sinceState = changesResponse.newStateChanges;
-
-        final newMailboxUpdated = await _combineMailboxCache(
-            mailboxUpdatedList: changesResponse.updated,
-            updatedProperties: changesResponse.updatedProperties,
-            mailboxCacheList: localMailboxResponse.mailboxes!);
-
-        await Future.wait([
-          mapDataSource[DataSourceType.local]!.update(
-              accountId,
-              session.username,
-              updated: newMailboxUpdated,
-              created: changesResponse.created,
-              destroyed: changesResponse.destroyed),
-          if (changesResponse.newStateMailbox != null)
-            stateDataSource.saveState(accountId, session.username, changesResponse.newStateMailbox!.toStateCache(StateType.mailbox)),
-        ]);
-      }
-    } else {
-      final mailboxResponse = await mapDataSource[DataSourceType.network]!.getAllMailbox(session, accountId);
-
-      await Future.wait([
-        mapDataSource[DataSourceType.local]!.update(accountId, session.username, created: mailboxResponse.mailboxes),
-        if (mailboxResponse.state != null)
-          stateDataSource.saveState(accountId, session.username, mailboxResponse.state!.toStateCache(StateType.mailbox)),
-      ]);
+    if (localMailboxResponse.hasMailboxes) {
+      log('MailboxRepositoryImpl::getAllMailbox: MAILBOX_CACHED = ${localMailboxResponse.mailboxes?.length} | STATE_CACHED = ${localMailboxResponse.state?.value}');
+      yield localMailboxResponse;
     }
 
+    final networkMailboxResponse = await mapDataSource[DataSourceType.network]!
```

Comment:
What happens if `mapDataSource[DataSourceType.local]!.getAllMailboxCache(accountId, session.username)` throw exception? please handle this case also